civis/ml/_model.py

```diff
@@ -819,6 +859,15 @@ def train(self, df=None, csv_path=None, table_name=None,
         polling_interval : float, optional
             Check for job completion every this number of seconds.
             Do not set if using the notifications endpoint.
+        validation_data : str, optional
+            Source for validation data. There are currently two options:
+            `'train'` (the default), which uses training data for validation;
+            and `'skip'`, which skips the validation step.
+        n_jobs : int, optional
+            Number of jobs to use for training and validation.
+        etl : Estimator, optional
+            Custom ETL estimator which overrides the default ETL, and
+            is run before training and validation.
```

Comment:
It seems like this should be part of the constructor. It's part of the model, and we set the model in the constructor.

Reply:
For that matter, `cross_validation_parameters` is also in the constructor. Almost everything is in the constructor. Should we put `validation_data` there too?

Reply:
If the selection were only between "train" and "skip", I'd say yes. But at some point we should add the option to input a new dataset. I think it's reasonable to keep it here. The `ModelPipeline` is not a scikit-learn Estimator, but its construction is similar. So the `train` parameters should be those which are data-dependent.

Reply:
Yeah, it wasn't clear to me how to choose where to put the arguments as I was writing the code, but that makes sense. Data-dependent parameters should go in the methods, parameters that can be treated as features of the model should go in the constructor.